tests/test_2025_08_20_mainnet_dg_upgrade.py

```diff
@@ -0,0 +1,423 @@
+        },
+    ]
+
+
+def test_vote(helpers, accounts, ldo_holder, vote_ids_from_env, stranger, dual_governance_proposal_calls):
+
+    # =======================================================================
+    # ========================= Arrange variables ===========================
+    # =======================================================================
+
+    dual_governance = interface.DualGovernance(DUAL_GOVERNANCE)
+    emergency_protected_timelock = interface.EmergencyProtectedTimelock(EMERGENCY_PROTECTED_TIMELOCK)
+    voting = interface.Voting(VOTING)
+    steth_token = interface.ERC20(STETH_TOKEN)
+
+    new_dual_governance = interface.DualGovernance(NEW_DUAL_GOVERNANCE)
+
+    # =========================================================================
+    # ======================== Identify or Create vote ========================
+    # =========================================================================
+
+    if len(vote_ids_from_env) > 0:
+        (vote_id,) = vote_ids_from_env
+        assert vote_id == EXPECTED_VOTE_ID
+    elif voting.votesLength() > EXPECTED_VOTE_ID:
+        vote_id = EXPECTED_VOTE_ID
+    else:
+        tx_params = {"from": LDO_HOLDER_ADDRESS_FOR_TESTS}
+        vote_id, _ = start_vote(tx_params, silent=True)
+
+    vote_script_onchain = voting.getVote(vote_id)["script"]
+    _, call_script_items = get_vote_items()
+    assert vote_script_onchain == encode_call_script(call_script_items)
+
+
+    # =========================================================================
+    # ============================= Execute Vote ==============================
+    # =========================================================================
+    is_voting_executed = voting.getVote(vote_id)["executed"]
+
+    if is_voting_executed:
+        print("Voting already executed. Skipping Aragon Voting phase tests.")
+
+    if not is_voting_executed:
+        # =======================================================================
+        # ========================= Before voting checks ========================
+        # =======================================================================
+
+        metadata = find_metadata_by_vote_id(vote_id)
+        assert get_lido_vote_cid_from_str(metadata) == IPFS_DESCRIPTION_HASH
+
+        assert interface.DGLaunchOmnibus(OMNIBUS_CONTRACT).isValidVoteScript(vote_id)
+
+        # =======================================================================
+        # ========================= Voting Execution ============================
+        # =======================================================================
+
+        vote_tx: TransactionReceipt = helpers.execute_vote(vote_id=vote_id, accounts=accounts, dao_voting=voting)
+
+        vote_events = group_voting_events_from_receipt(vote_tx)
+
+        assert EXPECTED_DG_PROPOSAL_ID == emergency_protected_timelock.getProposalsCount()
```

Comment:
Maybe it's worth to check that expected proposal id doesn't exist before the voting enactment?